sympy/functions/elementary/piecewise.py

```diff
@@ -172,11 +173,12 @@ def eval(cls, *_args):
         # make conditions canonical
         args = []
         for e, c in _args:
-            if not c.is_Atom and not isinstance(c, Relational):
+            if (not c.is_Atom and not isinstance(c, Relational) and not c.has(re)
+                and not c.has(im)):
```

Comment:
There could probably be `not c.has(re, im)`. (A single traversal instead of two.)

Reply:
Ahh, didn't know that worked. Will change it.

Reply:
Will still be two lines though, but better code.